### PR TITLE
icetouch 3.6.0

### PR DESCRIPTION
**Pull request by externl**
Add formula for Ice Touch 3.6.0

### CODE REVIEW

File: Library/Formula/icetouch.rb [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+  url "https://github.com/zeroc-ice/icetouch.git",
+    :tag => "v3.6.0", :revision => "f443edded03fac7f4e5c353ebc87efaf4bfa5229"
+
+  depends_on "mcpp"
+
+  def install
+    # Unset ICE_HOME as it interferes with the build
+    ENV.delete("ICE_HOME")
+    ENV.delete("USE_BIN_DIST")
+    ENV.delete("CPPFLAGS")
```
Comment (mistydemeo) [inline Library/Formula/icetouch.rb]: Does this not work with Homebrew's CPPFLAGS? This variable is set by Homebrew.

File: Library/Formula/icetouch.rb [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+    ENV.O2
+
+    args = %W[
```
Comment (mistydemeo) [inline Library/Formula/icetouch.rb]: If you're not modifying the argument array, you can just pass these as arguments directly to `make`.

File: Library/Formula/icetouch.rb (hunk 1 above)
Comment (externl) [inline Library/Formula/icetouch.rb]: Hi, yes it interferes with the icetouch build system.

File: Library/Formula/icetouch.rb (hunk 2 above)
Comment (externl) [inline Library/Formula/icetouch.rb]: Sure, I can pass these directly if you like.

Comment (mistydemeo) [inline Library/Formula/icetouch.rb]: Please!

File: Library/Formula/icetouch.rb (hunk 1 above)
Comment (mistydemeo) [inline Library/Formula/icetouch.rb]: How so? Can you specify?

Comment (externl) [inline Library/Formula/icetouch.rb]: I did some research, as this formula we've had around for a while. The issue was that ENV.02 was being overwritten by brew's CPPFLAGS. This seems to have been fixed though. I'll remove it from my formula.

**Comment by externl** [comment]
Pushed updates. Are bottles created automatically? Or do they have to be requested?

**Comment by DomT4** [comment]
> Are bottles created automatically?

Yup :+1:

**Comment by externl** [comment]
I'm not really sure what the failure was here.

**Comment by externl** [comment]
Looks like the git checkout failed. Is there a way to fix this?

```
==> brew install --verbose --build-bottle icetouch			 FAILED
==> Cloning https://github.com/zeroc-ice/icetouch.git
git --git-dir /Library/Caches/Homebrew/icetouch--git/.git status -s
Updating /Library/Caches/Homebrew/icetouch--git
git config remote.origin.url https://github.com/zeroc-ice/icetouch.git
git config remote.origin.fetch +refs/tags/v3.6.0:refs/tags/v3.6.0
git --git-dir /Library/Caches/Homebrew/icetouch--git/.git rev-parse -q --verify v3.6.0^{commit}
git checkout -f v3.6.0 --
HEAD is now at f443edd... Updated ice submodule to v3.6.0-rc3
git reset --hard v3.6.0
HEAD is now at f443edd Updated ice submodule to v3.6.0-rc3
git submodule foreach --recursive git submodule sync
git submodule update --init --recursive
Cloning into 'ice'...
Host key verification failed.
fatal: Could not read from remote repository.

Please make sure you have the correct access rights
and the repository exists.
Clone of 'git@github.com:zeroc-ice/ice' into submodule path 'ice' failed
Error: Failed to download resource "icetouch"
Failure while executing: git submodule update --init --recursive
```

**Comment by mistydemeo** [comment]
@BrewTestBot test this please

**Comment by externl** [comment]
I suspect this issue is caused by @BewtestBot not having an SSH key configured.

**Comment by externl** [comment]
I've fixed the repo to use https!

**Comment by externl** [comment]
Any update on getting this merged?

File: Library/Formula/icetouch.rb [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+    ENV.delete("ICE_HOME")
+    ENV.delete("USE_BIN_DIST")
+    ENV.O2
+
+    system "make", "install", "OPTIMIZE=yes", "prefix=#{prefix}"
+  end
+
+  test do
+    system "#{lib}/IceTouch/Cpp/bin/slice2cpp", "--version"
+    system "#{lib}/IceTouch/Objc/bin/slice2objc", "--version"
```
Comment (mikemcquaid) [inline Library/Formula/icetouch.rb]: Can this test be modified to do something more substantial than e.g. `--version` or `--help`? See `cmake.rb` for an example of an application formula with a good test and `tinyxml2.rb` for an example of a library formula with a good test. Thanks!

Comment (externl) [inline Library/Formula/icetouch.rb]: I've added a new test!

**Comment by externl** [comment]
```
==> audit problems
icetouch:
 * GitHub repository not notable enough (<10 forks, <10 watchers and <20 stars)
```
This was not an issue 12 days ago when I last updated this pull request.

**Comment by mikemcquaid** [comment]
> This was not an issue 12 days ago when I last updated this pull request.

Sorry but this is a result of us codifying our informal practises. Although this failure wasn't there we would probably have either rejected it or kicked ourselves after-the-fact for including it. I recommend making your own tap. Sorry!

**Pull request closed without merging**